frontend/test/pytest/test_mlir_plugin.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for standalone plugin"""
+
+from pathlib import Path
+
+import pennylane as qml
+
+from catalyst.passes import apply_pass, apply_pass_plugin
+from catalyst.utils.runtime_environment import get_bin_path
+
+plugin_path = get_bin_path("cli", "CATALYST_BIN_DIR") + "/../lib/StandalonePlugin.so"
+plugin = Path(plugin_path)
+
+
+def test_standalone_plugin():
+    """Generate MLIR for the standalone plugin. Do not execute code.
+    The code execution test is in the lit test. See that test
+    for more information as to why that is the case."""
+
+    @apply_pass("standalone-switch-bar-foo")
```

Comment:
Where is `--standalone-switch-bar-foo` registered? Is it innately accessible from mlir?

Reply:
https://github.com/PennyLaneAI/catalyst/pull/1317/commits/65245b246db56061d2892b9d4a0856b92be8cb35